common/build/eslint-config-fluid/scripts/print-configs.ts

```diff
@@ -0,0 +1,139 @@
+ */
+
+import fs from "node:fs/promises";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { ESLint } from "eslint";
+import sortJson from "sort-json";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+interface ConfigToPrint {
+	name: string;
+	configPath: string;
+	sourceFilePath: string;
+}
+
+const configsToPrint: ConfigToPrint[] = [
+	{
+		name: "default",
+		configPath: path.join(__dirname, "..", "index.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.ts"),
+	},
+	{
+		name: "minimal",
+		configPath: path.join(__dirname, "..", "minimal-deprecated.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.ts"),
+	},
+	{
+		name: "react",
+		configPath: path.join(__dirname, "..", "index.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.tsx"),
+	},
+	{
+		name: "recommended",
+		configPath: path.join(__dirname, "..", "recommended.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.ts"),
+	},
+	{
+		name: "strict",
+		configPath: path.join(__dirname, "..", "strict.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.ts"),
+	},
+	{
+		name: "strict-biome",
+		configPath: path.join(__dirname, "..", "strict-biome.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "file.ts"),
+	},
+	{
+		name: "test",
+		configPath: path.join(__dirname, "..", "recommended.js"),
+		sourceFilePath: path.join(__dirname, "..", "src", "test", "file.ts"),
+	},
+];
+
+/**
+ * Generates the applied ESLint config for a specific file and config path.
+ */
+async function generateConfig(filePath: string, configPath: string): Promise<string> {
+	console.log(`Printing config for ${filePath} using ${configPath}`);
+	const eslint = new ESLint({
+		overrideConfigFile: configPath,
+	});
+
+	const config = await eslint.calculateConfigForFile(filePath);
+	// Remove the parser property because it's an absolute path and will vary based on the local environment.
+	delete (config as { parser?: unknown }).parser;
+
+	// Generate the new content with sorting applied
+	// Sorting at all is desirable as otherwise changes in the order of common config references may cause large diffs
+	// with little semantic meaning.
+	// On the other hand, fully sorting the json can be misleading:
+	// some eslint settings depend on object key order ("import-x/resolver" being a known one, see
+	// https://github.com/un-ts/eslint-plugin-import-x/blob/master/src/utils/resolve.ts).
+	// Using depth 2 is a nice compromise.
+	const sortedConfig = sortJson(config, { indentSize: 4, depth: 2 });
+	const finalConfig = JSON.stringify(sortedConfig, null, 4);
+
+	// Add a trailing newline to match preferred output formatting
+	return finalConfig + "\n";
+}
+
+(async () => {
+	const args = process.argv.slice(2);
+
+	if (args.length !== 1) {
+		console.error("Usage: node print-configs.ts <output-directory>");
```

Comment:
? IIRC node did add support to run ts files directly at some point but it might still be experimental.
```suggestion
		console.error("Usage: tsx print-configs.ts <output-directory>");
```